src/migrations/20241024062521-add-release-plans-milestones-strategies.js

```diff
@@ -0,0 +1,62 @@
+exports.up = function(db, cb) {
+    db.runSql(
+        `
+        CREATE TABLE IF NOT EXISTS release_plan_definitions
+        (
+            id TEXT PRIMARY KEY NOT NULL,
+            discriminator TEXT NOT NULL,
+            name TEXT NOT NULL,
+            description TEXT,
+            feature_name TEXT,
+			environment TEXT,
+            created_by_user_id INTEGER NOT NULL REFERENCES users(id),
+            created_at TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now(),
+            CONSTRAINT release_plan_definitions_discriminator_values
+                CHECK (discriminator IN ('plan', 'template')),
+            CONSTRAINT feature_environments_fkey FOREIGN KEY (environment, feature_name)
+                REFERENCES feature_environments(environment, feature_name)
+                ON DELETE CASCADE
+        );
+
+        CREATE TABLE IF NOT EXISTS milestones
+        (
+            id SERIAL PRIMARY KEY NOT NULL,
+            name TEXT NOT NULL,
+            sort_order INTEGER NOT NULL,
+            release_plan_definition_id TEXT NOT NULL REFERENCES release_plan_definitions(id) ON DELETE CASCADE
+        );
+
+        CREATE INDEX IF NOT EXISTS idx_milestones_release_plan_definition_id ON milestones (release_plan_definition_id);
+
+        ALTER TABLE IF EXISTS release_plan_definitions
+            ADD COLUMN IF NOT EXISTS active_milestone_id SERIAL REFERENCES milestones(id);
```

Comment:
Certainly this is an INT (you don't want a separate sequence generated here)

Reply:
Yep 100% correct